packages/vue-app/template/router.js

```diff
@@ -3,16 +3,32 @@ import Router from 'vue-router'
 import { interopDefault } from './utils'
 
 <% function recursiveRoutes(routes, tab, components, indentCount) {
-  let res = ''
+  let res = '', resMap = ''
   const baseIndent = tab.repeat(indentCount)
   const firstIndent = '\n' + tab.repeat(indentCount + 1)
   const nextIndent = ',' + firstIndent
   routes.forEach((route, i) => {
-    route._name = '_' + hash(route.component)
-    components.push({ _name: route._name, component: route.component, name: route.name, chunkName: route.chunkName })
+    if(route.components) {
+      let _name = '_' + hash(route.components.default)
+      resMap += firstIndent + tab + 'default: ' + (splitChunks.pages ? _name : `() => ${_name}.default || ${_name}`)
+      Object.keys(route.components).map(function(k){
+        _name = '_' + hash(route.components[k])
+        if (k === 'default') {
+          components.push({_name: _name, component: route.components[k], name: route.name, chunkName: route.chunkName})
+        } else {
+          components.push({_name: _name, component: route.components[k], name: route.name + '-' + k, chunkName: route.chunkNames[k]})
+          resMap += nextIndent + tab + k + ': ' + (splitChunks.pages ? _name : `() => ${_name}.default || ${_name}`)
+        }
```

Comment:
@karakum Following the same line of thought:

```js
        const cmpnnt = { _name, component: route.components[k] }
        if (k === 'default') {
          components.push({
          	...cmpnnt,
          	name: route.name,
          	chunkName: route.chunkName
          	})
        } else {
          components.push({
          	...cmpnnt,
          	name: `${route.name}-${k}`,
          	chunkName: route.chunkNames[k]
          })
          if (splitChunks.pages) {
          	resMap += `${nextIndent}${tab}${k}: ${_name}`
          } else {
          	resMap += `${nextIndent}${tab}${k}: ` + `() => ${_name}.default || ${_name}`
          }
        }
```

Anyway, just thought I'd encourage structuring code for readability. 

But PR looks great as-is. Cheers!